pkg/types/cciptypes/models.go

```diff
@@ -0,0 +1,38 @@
+package cciptypes
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type Address string
+
+func (a Address) Equals(addr2 Address) bool {
+	if common.IsHexAddress(string(a)) && common.IsHexAddress(string(addr2)) {
+		return common.HexToAddress(string(a)) == common.HexToAddress(string(addr2))
+	}
+	return a == addr2
+}
+
+func (a Address) ToEVM() (common.Address, error) {
+	if !common.IsHexAddress(string(a)) {
+		return common.Address{}, fmt.Errorf("%s is not a hex evm address", a)
+	}
+	return common.HexToAddress(string(a)), nil
+}
+
+type Hash [32]byte
```

Comment:
@jmank88 do you have an advise here for generalizing the ccip types? specific do you know we already a common definition of a hash that is known to apply to various chains? my concern is that constraining to 32 bytes might be too limiting, but i'm not sure

Reply:
Yeah if the length assumption can be lifted, that would be helpful. Another angle would be to pass around a human readable string encoding, like with `types.Account`.